pyproject.toml

```diff
@@ -36,6 +37,8 @@ ignore = [
     "D203",
     "D213",
     "S101",
+    "ARG002",
```

Comment:
I think if we create functions that do not use some arguments, then our function is not ok. We should ignore functions inline instead

Reply:
Let's just forget about that. Mypy still fails to validate any cls type annotation alongside with returning Self for some reason 🤷‍♂️